common/src/test/resources/ssl/generate_certs.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A simple utility to (re-)generate the files within resources/ssl
+# These generated certificates are used for a variety of test scenarios for SSL.
+# The utility ends up generating two certificates - which are saved into two different keystores
+# The certificates generated are signed by two different CA cert's (also generated here).
+# There are two truststores generated - the first truststore has both CA certs as part of it
+# Hence this trust can be used to validate both client certificates.
+# The second trust store has NO CA certs in it - and so when used will fail both the certificates.
+
+function gen_certs() {
+
+  openssl genrsa -out ca.key 2048
```

Comment:
Seems it requires installing some OpenSSL-related software (I saw some Linux distro drop OpenSSL 1.1 in APT repo, only keeps OpenSSL 3, does it matter?) before using this script, could you please add a brief comment to guide how to set up the environment before running this script? For example, provide an Ubuntu docker-based example.

Reply:
This script is provided in case the files need to be regenerated - the script is not used within the codebase.
Otherwise, it will be some binary that got committed without a way to (re)generate them.

Reply:
I know that, guidance for setup environment is still required in such case.